src/pages/api/modules/dns-hole/summary.ts

```diff
@@ -26,21 +30,46 @@ export const Get = async (request: NextApiRequest, response: NextApiResponse) =>
     const app = applicableApps[i];
 
     try {
-      const piHole = new PiHoleClient(app.url, findAppProperty(app, 'password'));
-
-      // eslint-disable-next-line no-await-in-loop
-      const summary = await piHole.getSummary();
-
-      data.domainsBeingBlocked += summary.domains_being_blocked;
-      data.adsBlockedToday += summary.ads_blocked_today;
-      data.dnsQueriesToday += summary.dns_queries_today;
-      data.status.push({
-        status: summary.status,
-        appId: app.id,
-      });
-      adsBlockedTodayPercentageArr.push(summary.ads_percentage_today);
+      switch (app.integration?.type) {
```

Comment:
Maybe you should check if `app.integration` exists before starting the loop, and return and error if it doesn't

Reply:
We already do. Check line 16. I added a default case which will log an error.